_fix_string_length clips last character if string length >= size

If the string being passed to  `_fix_string_length` is equal to or longer than the `size` argument, the last character gets clipped off.
For example, if I run the test_slots_and_tokens unit test with a token label of 32 characters, it fails like so:

### Command
```
PKCS11_TOKEN_SO_PIN=654321 PKCS11_TOKEN_PIN=123456 PKCS11_TOKEN_LABEL="73ef7a32521d4b568c462281e6a9ac4c" PKCS11_MODULE='libsofthsm2.so' python3 -m unittest -v tests.test_slots_and_tokens
```
### Error
```
======================================================================
FAIL: test_get_token (tests.test_slots_and_tokens.SlotsAndTokensTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "python-pkcs11/tests/test_slots_and_tokens.py", line 72, in test_get_token
    self.assertEqual(token.label, TOKEN)
AssertionError: '73ef7a32521d4b568c462281e6a9ac4' != '73ef7a32521d4b568c462281e6a9ac4c'
- 73ef7a32521d4b568c462281e6a9ac4
+ 73ef7a32521d4b568c462281e6a9ac4c
?                                +
```
It looks like this is because the `_fix_string_length` function is adding a null character as position `size - 1`; I think this should just be `string[size] = b'\0'`. Because pkcs11 pads strings less than the maximum size, this doesn't appear to cause a problem until the string is at least the same length as the `size` argument, whereupon the last character is clipped off.

So, I can now pass the test test_slots_and_tokens test with a label of 32 characters. However, I'm getting failures on the test_x509 and test_ecc cases (both on my branch, and on master):

### test_x509
#### Test Output
```
test_import_ca_certificate (tests.test_x509.X509Tests) ... ok
test_import_ca_certificate_easy (tests.test_x509.X509Tests) ... ok
test_self_sign_certificate (tests.test_x509.X509Tests) ... CN = Test Certificate
error 18 at 0 depth lookup: self signed certificate
FAIL
test_sign_csr (tests.test_x509.X509Tests) ... verify OK
ok
test_verify_certificate_dsa (tests.test_x509.X509Tests) ... ok
test_verify_certificate_ecdsa (tests.test_x509.X509Tests) ... skipped 'Requires Mechanism.ECDSA_SHA1'
test_verify_certificate_rsa (tests.test_x509.X509Tests) ... ok

======================================================================
FAIL: test_self_sign_certificate (tests.test_x509.X509Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/aidan/work/python-pkcs11/tests/__init__.py", line 95, in wrapper
    return func(self, *args, **kwargs)
  File "/home/aidan/work/python-pkcs11/tests/test_x509.py", line 222, in test_self_sign_certificate
    self.assertEqual(proc.wait(), 0)
AssertionError: 2 != 0

----------------------------------------------------------------------
Ran 7 tests in 0.094s

FAILED (failures=1, skipped=1)
Test failed: <unittest.runner.TextTestResult run=7 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=7 errors=0 failures=1>
```
### test_ecc
#### Test Output
```
test_derive_key (tests.test_ecc.ECCTests) ... ERROR
test_import_key_named_curve (tests.test_ecc.ECCTests) ... skipped 'Requires Mechanism.ECDSA_SHA1'
test_import_key_pair (tests.test_ecc.ECCTests) ... ok
test_import_key_params (tests.test_ecc.ECCTests) ... ok
test_sign_ecdsa (tests.test_ecc.ECCTests) ... ok
test_sign_eddsa (tests.test_ecc.ECCTests) ... ERROR

======================================================================
ERROR: test_derive_key (tests.test_ecc.ECCTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/aidan/work/python-pkcs11/tests/__init__.py", line 95, in wrapper
    return func(self, *args, **kwargs)
  File "/home/aidan/work/python-pkcs11/tests/test_ecc.py", line 43, in test_derive_key
    alice_pub, alice_priv = parameters.generate_keypair()
  File "pkcs11/_pkcs11.pyx", line 740, in pkcs11._pkcs11.DomainParameters.generate_keypair
    assertRV(_funclist.C_GenerateKeyPair(self.session._handle,
  File "pkcs11/_errors.pyx", line 88, in pkcs11._pkcs11.assertRV
    raise ERROR_MAP.get(rv,
pkcs11.exceptions.GeneralError

======================================================================
ERROR: test_sign_eddsa (tests.test_ecc.ECCTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/aidan/work/python-pkcs11/tests/__init__.py", line 95, in wrapper
    return func(self, *args, **kwargs)
  File "/home/aidan/work/python-pkcs11/tests/test_ecc.py", line 159, in test_sign_eddsa
    pub, priv = parameters.generate_keypair()
  File "pkcs11/_pkcs11.pyx", line 740, in pkcs11._pkcs11.DomainParameters.generate_keypair
    assertRV(_funclist.C_GenerateKeyPair(self.session._handle,
  File "pkcs11/_errors.pyx", line 88, in pkcs11._pkcs11.assertRV
    raise ERROR_MAP.get(rv,
pkcs11.exceptions.GeneralError

----------------------------------------------------------------------
Ran 6 tests in 0.013s

FAILED (errors=2, skipped=1)
Test failed: <unittest.runner.TextTestResult run=6 errors=2 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=6 errors=2 failures=0>
```
These are errors from the Journal when running test_ecc:
```
python3[25382]: OSSLECDSA.cpp(304): ECDSA key generation failed (0x100AE081)
python3[25382]: SoftHSM.cpp(8689): Could not generate key pair
python3[25382]: OSSLECDSA.cpp(304): ECDSA key generation failed (0x10108077)
python3[25382]: SoftHSM.cpp(8689): Could not generate key pair
```

Do you have any thoughts on these errors?

--- a/pkcs11/_pkcs11.pyx
+++ b/pkcs11/_pkcs11.pyx
@@ -192,12 +192,13 @@ class Slot(types.Slot):
 
         assertRV(_funclist.C_GetTokenInfo(self.slot_id, &info))
 
-        _fix_string_length(info.label, sizeof(info.label))
-        _fix_string_length(info.serialNumber, sizeof(info.serialNumber))
-        _fix_string_length(info.manufacturerID, sizeof(info.manufacturerID))
-        _fix_string_length(info.model, sizeof(info.model))
+        label = info.label[:sizeof(info.label)]
+        serialNumber = info.serialNumber[:sizeof(info.serialNumber)]
+        model = info.model[:sizeof(info.model)]
+        manufacturerID = info.manufacturerID[:sizeof(info.manufacturerID)]
 
-        return Token(self, **info)
+        return Token(self, label, serialNumber, model, manufacturerID,
+                     info.hardwareVersion, info.firmwareVersion, info.flags)
 
     def get_mechanisms(self):
         cdef CK_ULONG count
@@ -1247,13 +1248,11 @@ cdef class lib:
         cdef CK_INFO info
         assertRV(_funclist.C_GetInfo(&info))
 
-        _fix_string_length(info.manufacturerID,
-                           sizeof(info.manufacturerID))
-        _fix_string_length(info.libraryDescription,
-                           sizeof(info.libraryDescription))
+        manufacturerID = info.manufacturerID[:sizeof(info.manufacturerID)]
+        libraryDescription = info.libraryDescription[:sizeof(info.libraryDescription)]
 
-        self.manufacturer_id = _CK_UTF8CHAR_to_str(info.manufacturerID)
-        self.library_description = _CK_UTF8CHAR_to_str(info.libraryDescription)
+        self.manufacturer_id = _CK_UTF8CHAR_to_str(manufacturerID)
+        self.library_description = _CK_UTF8CHAR_to_str(libraryDescription)
         self.cryptoki_version = _CK_VERSION_to_tuple(info.cryptokiVersion)
         self.library_version = _CK_VERSION_to_tuple(info.libraryVersion)
 
@@ -1291,12 +1290,13 @@ cdef class lib:
         for slotID in slotIDs:
             assertRV(_funclist.C_GetSlotInfo(slotID, &info))
 
-            _fix_string_length(info.slotDescription,
-                               sizeof(info.slotDescription))
-            _fix_string_length(info.manufacturerID,
-                               sizeof(info.manufacturerID))
+            slotDescription = info.slotDescription[:sizeof(info.slotDescription)]
+            manufacturerID = info.manufacturerID[:sizeof(info.manufacturerID)]
 
-            slots.append(Slot(self, slotID, **info))
+            slots.append(
+                Slot(self, slotID, slotDescription, manufacturerID,
+                     info.hardwareVersion, info.firmwareVersion, info.flags)
+            )
 
         return slots
 

--- a/pkcs11/_utils.pyx
+++ b/pkcs11/_utils.pyx
@@ -36,8 +36,3 @@ cdef _unpack_attributes(key, value):
     except KeyError:
         raise NotImplementedError("Can't unpack this %s. "
                                   "Expand ATTRIBUTE_TYPES!" % key)
-
-
-cdef _fix_string_length(CK_UTF8CHAR *string, size_t size):
-    """Insert NUL-bytes into strings so Python knows their length."""
-    string[size - 1] = b'\0'

--- a/tests/test_aes.py
+++ b/tests/test_aes.py
@@ -107,7 +107,6 @@ class AESTests(TestCase):
 
     @requires(Mechanism.AES_KEY_WRAP)
     @FIXME.opencryptoki  # can't set key attributes
-    @FIXME.travis  # Travis has an old OpenSSL
     def test_wrap(self):
         key = self.session.generate_key(pkcs11.KeyType.AES, 128, template={
             pkcs11.Attribute.EXTRACTABLE: True,

